icic_extraction/grid.py

```diff
@@ -0,0 +1,84 @@
+import re 
+import pandas as pd
+import pdfplumber
+def get_text_from_pdf_with_plumber(apdf):
+    tot = ''
+    with pdfplumber.open(apdf) as pdf:
+        for p in pdf.pages:
+            tot += "\n"+p.extract_text()
+    return tot 
+def get_meta_from_contract(contract_cont):
+    contract_note_no = re.findall("Contract Note No.*?:(.*)",contract_cont)[0]
+    trade_date = re.findall ("TRADE DATE\s*?:\s*(\d{2}-\d{2}-\d{4}).*",contract_cont)[0]
+    settlemnt_no = re.findall ("SETTLEMENT NO.*:\s*(\d+).*",contract_cont)[0]
+    settlement_date = re.findall ("SETTLEMENT DATE.*:\s*(\d{2}-\d{2}-\d{4}).*",contract_cont)[0]
+    mob_num = re.findall ("Mob No:\s*(\d*).*",contract_cont)[0]
+    ucc_no = re.findall("UCC of Client\s*:\s*(\d*)",contract_cont)[0]
+    pan_of_client = re.findall ("PAN of Client\s*:\s*(.*)",contract_cont)[0]
+    meta_contract = [contract_note_no,trade_date,settlemnt_no,settlement_date,mob_num,ucc_no,pan_of_client]
+    return meta_contract
+def get_summary_table(contract_cont):
+    seperate_summary_table = re.findall("Transaction Tax(.*?)\s*Brokerage Charges Charges.*?Buy(.*?)Sell(.*?)Total Payable / Receivable(.*?)\n",contract_cont,re.DOTALL)
+    return seperate_summary_table
+def get_detailed_table (contract_cont):
+    seperate_detailed_table = re. findall("\d+\s+\d+\s+\d+\s+[\d{2}:\d{2}].*",contract_cont)
+    return seperate_detailed_table
+def extract_transaction_from_detailed_table(detailed_table):
+    transactions=[]   
+    for record in detailed_table :
+        records_list = record.split()
+        name = " ".join(records_list[5:len(records_list)-7])
+        del records_list[5:len(records_list)-7]
+        records_list.insert(5,name)
+        transactions.append(records_list)
+    return transactions
+def extract_transaction_from_summary_table (summary_table):
+    summary_table_data = []
+    for transaction  in summary_table:
+        clean_transaction = tuple(element.strip() for element in transaction)
+        clean_transaction_lis = list(clean_transaction)
+        company_name = clean_transaction_lis.pop(0)
+        transaction_record = [x.split() for x in clean_transaction_lis ]
+        transaction_record.insert(0,company_name)
+        summary_table_data.append(transaction_record)
+    return summary_table_data
+def get_meta_table(contract_cont):
```

Comment:
better naming as we consider the meta -> contract note no, client name and so on 
naming suggestion : get_totals_table

Reply:
was this done?